apps/kol-amelamdim/pages/category/[category].tsx

```diff
@@ -0,0 +1,174 @@
+  const { isAuthenticated } = useContext(AuthContext);
+  const { setAlertMessage, setAlertType } = useContext(AlertContext);
+
+  //TODO: handle Error & loading & no data to show.
+  //TODO: handle file upload to be added to the tabled immediately
+  const handleChangePage = (event: unknown, newPage: number) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (event: ChangeEvent<HTMLInputElement>) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
+
+  const handleFilter = () => {
+    let filteredFileType = [...files];
+    if (
+      fileType &&
+      FILE_TYPES_DICTIONARY[fileType] !== FILE_TYPES_DICTIONARY[10]
+    ) {
+      filteredFileType = filteredFileType.filter(
+        (file) => file.type === FILE_TYPES_DICTIONARY[fileType]
+      );
+    }
+
+    let filteredTextData = [...filteredFileType];
+    if (filterText) {
+      filteredTextData = filteredTextData.filter(
+        (file) =>
+          file.name.includes(filterText) || file.author.includes(filterText)
+      );
+    }
+
+    setFilteredFiles(filteredTextData);
+  };
+
+  const handleShareContentButtonClick = async () => {
+    if (isAuthenticated) {
+      setIsUploadFileDialogOpen(true);
+    } else {
+      await router.push('/login');
+    }
+  };
+  useEffect(() => {
+    if (files) {
+      setFilteredFiles(files);
+    }
+  }, [files]);
+
+  useEffect(() => {
+    if (error) {
+      setAlertType('warning');
+      setAlertMessage(API_ERRORS.errorFetchData.message.heb);
+    }
+  }, [error]);
+
+  const renderNoData = () =>
+    loading ? (
+      <div style={{ fontSize: '200px' }}>טוען..</div>
+    ) : (
+      <div>אין מידע להציג</div>
+    );
+  return (
+    <StyledPageContainer>
+      {loading ? (
+        renderNoData()
+      ) : (
+        <>
+          <FilterCard
+            setFileType={setFileType}
+            fileType={fileType}
+            filterText={filterText}
+            setFilterText={setFilterText}
+            onClick={handleFilter}
+          />
+          <TableContainer component={Paper}>
+            <Table sx={{ minWidth: 650 }}>
+              <TableHead>
+                <TableRow>
+                  <TableCell>שם</TableCell>
+                  <TableCell>מחבר</TableCell>
+                  <TableCell>גודל קובץ (מגה בייט)</TableCell>
+                  <TableCell>סוג קובץ</TableCell>
+                  <TableCell>לינק להורדה</TableCell>
+                </TableRow>
+              </TableHead>
+              <TableBody>
+                {filteredFiles
+                  .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
+                  .map((row) => {
+                    return (
+                      <TableRow key={row.key}>
+                        <TableCell>{row.name}</TableCell>
+                        <TableCell>{row.author}</TableCell>
+                        <TableCell>{row.size}</TableCell>
```

Comment:
we should format the size from some function the live in utils

Reply:
Resolved.